src/schedule/ucc_schedule.h

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+#ifndef UCC_SCHEDULE_H_
```

Comment:
Given that this is core and basis for all collectives, we should document it.

Reply:
i'll add basic description